### PR TITLE
[Framework][API] Remove the deprecated gas_currency_code field

### PR DESCRIPTION
**Pull request by movekevin**
### Description
This is an artifact from Diem time and no longer relevant - Aptos only has one gas currency (APT).

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2902)

### CODE REVIEW

File: aptos-move/aptos-vm/src/aptos_vm_impl.rs [hunk 1]
```diff
@@ -285,7 +284,8 @@ impl AptosVMImpl {
             .execute_function_bypass_visibility(
                 &chain_specific_info.module_id(),
                 prologue_function_name,
-                gas_currency,
+                // TODO: Deprecate this once we remove gas currency on the Move side.
+                vec![],
```
Comment (gregnazario) [inline aptos-move/aptos-vm/src/aptos_vm_impl.rs:288]: Have we made the move change yet?

File: aptos-move/aptos-vm/src/errors.rs [hunk 2]
```diff
@@ -26,6 +26,8 @@ pub const ESCRIPT_NOT_ALLOWED: u64 = 1008;
 pub const EMODULE_NOT_ALLOWED: u64 = 1009;
 pub const EINVALID_WRITESET_SENDER: u64 = 1010; // invalid sender (not aptos root) for write set
 pub const ESEQUENCE_NUMBER_TOO_BIG: u64 = 1011;
+// This error code has been deprecated and just left here sp future error codes would not
+// accidentally reuse.
 pub const EBAD_TRANSACTION_FEE_CURRENCY: u64 = 1012;
```
Comment (gregnazario) [inline aptos-move/aptos-vm/src/errors.rs:31]: Maybe we should rename this to RESERVED_UNUSED or something?

File: ecosystem/indexer-server/doc/openapi.yaml [hunk 3]
```diff
@@ -726,7 +726,7 @@ components:
         Note:
           1. Empty chars should be ignored when comparing 2 struct tag ids.
           2. When used in an URL path, should be encoded by url-encoding (AKA percent-encoding).
-      example: "0x1::XUS::XUS"
+      example: "0x1::aptos_coin::AptosCoin"
```
Comment (gregnazario) [inline ecosystem/indexer-server/doc/openapi.yaml:729]: Why do we have a copy of the open API spec here from the API @CapCap ?

File: ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml [hunk 4]
```diff
@@ -1155,7 +1155,7 @@ components:
     AccountResource:
       description: Account resource is a Move struct value belongs to an account.
       example:
-        type: 0x1::AptosAccount::Balance<0x1::XDX::XDX>
+        type: 0x1::coin::CoinStore<0x1::XDX::XDX>
```
Comment (gregnazario) [inline ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml:1158]: This file is supposed to be autogenerated and should use the other spec

**Review by gregnazario — APPROVED** [review]
@bowenyang007 @CapCap 

Let's cleanup the indexer server to use the already existing Open API specs.

@movekevin Is the gas code already removed from Move / can we even do that.  If we can't then please remove the associated comments.

File: aptos-move/aptos-vm/src/aptos_vm_impl.rs (hunk 1 above)
Comment (movekevin) [reply aptos-move/aptos-vm/src/aptos_vm_impl.rs:288]: Nope that's why I'm keeping vec![] here. I'll look into the change on Move side separately

File: ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml (hunk 4 above)
Comment (movekevin) [reply ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml:1158]: Are there instructions or regenerating?

File: aptos-move/aptos-vm/src/errors.rs (hunk 2 above)
Comment (movekevin) [reply aptos-move/aptos-vm/src/errors.rs]: Updated

File: ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml (hunk 4 above)
Comment (gregnazario) [reply ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml:1158]: The new spec has none of this, so once the old one is deleted you don't have to do anything!

Comment (movekevin) [reply ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml]: this is autogenerated but still checked into the repo though. I don't see it being updated automatically when running build or lint

Comment (gregnazario) [reply ecosystem/indexer-server/typescript/src/api/v1/openapi.yaml]: It's outside the scope of this.  It's supposed to regenerate automatically or tell you in the CI if it wasn't updated.

Either way the currency code isn't even in the spec to be generated so there is no change in the autogenerated code.

**Pull request merged**